tests/rest/media/v1/test_media_storage.py

```diff
@@ -201,36 +202,49 @@ class _TestImage:
     ],
 )
 class MediaRepoTests(unittest.HomeserverTestCase):
-
+    test_image: ClassVar[_TestImage]
     hijack_auth = True
     user_id = "@test:user"
 
     def make_homeserver(self, reactor: MemoryReactor, clock: Clock) -> HomeServer:
 
-        self.fetches = []
+        self.fetches: List[
+            Tuple[
+                "Deferred[Tuple[bytes, Tuple[int, Dict[bytes, List[bytes]]]]]",
+                str,
+                str,
+                Optional[QueryParams],
+            ]
+        ] = []
 
         def get_file(
             destination: str,
             path: str,
             output_stream: BinaryIO,
-            args: Optional[Dict[str, Union[str, List[str]]]] = None,
+            args: Optional[QueryParams] = None,
+            retry_on_dns_fail: bool = True,
             max_size: Optional[int] = None,
-        ) -> Deferred:
-            """
-            Returns tuple[int,dict,str,int] of file length, response headers,
-            absolute URI, and response code.
-            """
+            ignore_backoff: bool = False,
+        ) -> Deferred[Tuple[int, Dict[bytes, List[bytes]]]]:
```

Comment:
This doesn't match:

```suggestion
        ) -> Deferred[Tuple[int, Dict[bytes, List[bytes]], str, int]]:
```

Reply:
I think the original docstring was out of date. As written, it does match the signature of MatrixFederationHttpClient.get_file

Reply:
`(╯°□°）╯︵ ┻━┻`

`MatrixFederationHttpClient.get_file` and `SimpleHttpClient.get_file` have different signatures.